src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+                    String at = eventArray[1];
+                    taskArrayList.add(new Event(description, at, isDone));
+                    break;
+                }
+
+            }
+        } else {
+            File f = new File (dataPath);
+            boolean dirExists = f.mkdir();
+            if (dirExists) {
+                f = new File (dataPath + "/tasks.txt");
+                f.createNewFile();
+            } else {
+                throw new DukeException();
+            }
+        }
+    }
+
+    public static void writeFile() throws IOException, DukeException {
+        String dataPath = new File("saved_data/tasks.txt").getAbsolutePath();
+        StringBuilder fullString = new StringBuilder();
+
+        for(Task task : taskArrayList) {
+            if (task instanceof ToDo) {
+                ToDo todo = (ToDo) task;
+                String todoString = todo.getTypeIcon() + "|" + todo.isDone + "|"
+                        + todo.description + System.lineSeparator();
+                fullString.append(todoString);
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                String eventString = event.getTypeIcon() + "|" + event.isDone + "|"
+                        + event.description + "|" + event.at + System.lineSeparator();
+                fullString.append(eventString);
+            } else if (task instanceof Deadline) {
+                Deadline deadline = (Deadline) task;
+                String deadlineString = deadline.getTypeIcon() + "|" + deadline.isDone + "|"
+                        + deadline.description + "|" + deadline.by + System.lineSeparator();
+                fullString.append(deadlineString);
+            } else {
+                throw new DukeException();
+            }
+        }
+
+        FileWriter fw = new FileWriter(dataPath);
+        fw.write(String.valueOf(fullString));
+        fw.close();
+    }
+
+    public static void main(String[] args) {
+
+        System.out.println(format);
+        System.out.println("Hey I'm Butler!\n");
+        System.out.println("How can I help you Monsieur/Madame?\n");
+        System.out.println(format);
+        String[] stringArray;
+
+        try {
+            readFile();
+        } catch (IOException | DukeException e) {
+            System.out.println("There was an error reading saved data!");
+        }
+
+        do {
+
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+
+            stringArray = line.trim().split(" ", 2);
+            try {
+                parseInput(stringArray);
+            } catch (EmptyDescriptionException e){
+                e.displayException();
+            } catch (WrongCommandException e){
+                e.displayException();
+            } catch (EmptyDateException e){
+                e.displayException();
+            }
+            
+        } while (!stringArray[0].equals("bye"));
```

Comment:
variable naming can be improved. Also string literals like "bye" can be declared as named constants instead.